Atomic/bond targets prediction (#280)
chemprop/chemprop: README.md

  * `--depth_solvent` Number of message passing steps for solvent/molecule.

>   * `--adding_h` Whether RDKit molecules will be constructed with adding the Hs to them. Applicable to any SMILES that is not reaction.

Author: Yes! That is a good point. Thanks!

### Atomic and bond properties prediction

> Chemprop can perform multitask constraint message passing neural networks for atomic/bond properties prediction as described in this [paper](https://chemrxiv.org/articles/preprint/Regio-Selectivity_Prediction_with_a_Machine-Learned_Reaction_Representation_and_On-the-Fly_Quantum_Mechanical_Descriptors/12907316). This model can train on any number of atomic/bond properties simultaneously. In the original work, a total loss was calculated as a weighted sum of every single loss, where the weights were required to be specified for the regression task. In this repository, these weights have been automatically taken into account by doing standardization of all the training targets. In order to train a model, training data containing molecules (as SMILES strings) and known atomic/bond target values are required. The input format can either be a csv or pickle file. For example:

Reviewer: ```suggestion
Chemprop can perform multitask constrained message passing neural networks for atomic/bond properties prediction as described in this [paper](https://chemrxiv.org/articles/preprint/Regio-Selectivity_Prediction_with_a_Machine-Learned_Reaction_Representation_and_On-the-Fly_Quantum_Mechanical_Descriptors/12907316). This model can train on any number of atomic/bond properties simultaneously. In the original work, a total loss was calculated as a weighted sum of every single loss, where the weights were required to be specified for the regression task. In this repository, these weights have been automatically taken into account by doing standardization of all the training targets. In order to train a model, training data containing molecules (as SMILES strings) and known atomic/bond target values are required. The input format can either be a csv or pickle file. For example:
```

4      CC(=N)NCc1cccc(CNCc2ccncc2)c1  [-0.083162, 0.114954, -0.274544, -0.100369, 0....  ...  [[0.0, 1.5137126697008916, 2.4882198180715465,...  [[0.0, 1.0036, 0.0437, 0.0108, 0.0134, 0.0004,......
```
where atomic properties (e.g. hirshfeld_charges) must be a 1D list with the order same as that of atoms in the SMILES string; and bond properties (e.g. bond_length_matrix) can either be a 2D list of shape (number_of_atoms × number_of_atoms) or a 1D list with the order same as that of bonds in the SMILES string. If the input format is a pickle file, these targets can also be saved as numpy arrays.
The atomic/bond training targets can be assigned by using `--atom_targets` and `--bond_targets`. This model allows multitask constraints applied to different atomic/bond properties by specifying the argument `--constraints_path` with a given `.csv` file. Note that the constraints must be in the same order as the SMILES strings in your data file. Also note that `.csv` file must have a header row and the constraints should be comma-separated with one line per molecule. If the 

> 

Reviewer: Seems like the description in L318 is accidentally unfinished.

Reviewer: If you provide constraints for one task, do you have to provide constraints for all tasks? What happens if you have blanks in the constraints file? Also probably worth mentioning that only atom-targets can be constrained.

Reviewer: Oh, I see below that it chooses if tasks are constrained based off of the header. I don't think there is otherwise an error that triggers if there are multiple copies of a task name in a header. You might need to add that to avoid this edge case.

> This model allows multitask constraints applied to different atomic/bond properties by specifying the argument `--constraints_path` with a given `.csv` file. Note that the constraints must be in the same order as the SMILES strings in your data file. Also note that `.csv` file must have a header row and the constraints should be comma-separated with one line per molecule. The `--adding_h` option can be used if hydrogens are included in the atoml targets and bonds to hydrogens are included in the bond targets.

Reviewer: Looks like I left a typo in my suggestion, sorry

```suggestion
This model allows multitask constraints applied to different atomic/bond properties by specifying the argument `--constraints_path` with a given `.csv` file. Note that the constraints must be in the same order as the SMILES strings in your data file. Also note that `.csv` file must have a header row and the constraints should be comma-separated with one line per molecule. The `--adding_h` option can be used if hydrogens are included in the atom targets and bonds to hydrogens are included in the bond targets.
```

Author: No problem! I missed this typo XD

> The atomic/bond training targets can be assigned by using `--atom_targets` and `--bond_targets`. This model allows multitask constraints applied to different atomic/bond properties by specifying the argument `--constraints_path` with a given `.csv` file. Note that the constraints must be in the same order as the SMILES strings in your data file. Also note that `.csv` file must have a header row and the constraints should be comma-separated with one line per molecule. The `--adding_h` option can be used if hydrogens are considered in SMILES.

Reviewer: ```suggestion
The atomic/bond training targets can be assigned by using `--atom_targets` and `--bond_targets`. This model allows multitask constraints applied to different atomic/bond properties by specifying the argument `--constraints_path` with a given `.csv` file. Note that the constraints must be in the same order as the SMILES strings in your data file. Also note that `.csv` file must have a header row and the constraints should be comma-separated with one line per molecule. The `--adding_h` option can be used if hydrogens are included in the atoml targets and bonds to hydrogens are included in the bond targets.
```
